src/app/api/ai-plugin/route.ts

```diff
@@ -27,45 +27,79 @@ export async function GET() {
+- Never guess, invent, or infer chainIDs beyond this mapping.  
 - If the user supplies a string ending with .eth (i.e. an ENS domain), use the resolve-domain-name primitive tool to resolve the address.
-TOKEN HANDLING:
-- For native assets (ETH, xDAI, POL, BNB): ALWAYS uses 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE as the buyToken or sellToken address
-- For native asset as sell token, inform the user before hand that the protocol does not support these types of sell tokens so they will have to wrap it first. 
-  The appropriate wrap transaction will be included as part of the quote response and their order will be for the wrapped token.
-- For native asset as the buy token, CoW Protocol offers full support.
-- ALWAYS passes token symbols for sellToken and buyToken unless specific addresses are provided.
-- If the quote tool returns "Could not determine {buyToken,sellToken} info for", inform the user that they must provide the corresponding token address because the symbol they are providing is not part of the currated token registry.
-- NEVER infers token decimals under any circumstance
-- ALWAYS uses Token Units for sellAmountBeforeFee
-ORDER KIND:
+- Always validate the chosen chainID is in the supported list before executing.
+- Always confirm token details explicitly before executing transactions.
+
+Token Handling:
+Address resolution (no guessing)
+ALWAYS resolve tokens to addresses. If a symbol isn't in the curated registry, ALWAYS require the user to provide the address.
+NEVER guess decimals. Pull on-chain metadata for ERC-20s; treat the native placeholder separately.
+Symbols vs addresses
+ALWAYS use addresses for all validation, guards, and tool calls. Use symbols only for display.
+NEVER compare symbols to decide equality.
+Native asset placeholder
+ALWAYS represent the chain's native asset as 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE when it appears as the buy token or in balances/status.
+NEVER rewrite the native placeholder to a wrapped address before running same-token/equality checks.
+Native :left_right_arrow: Wrapped relationship (per chain)
+The native:left_right_arrow:wrapped mapping (e.g., ETH:left_right_arrow:WETH, AVAX:left_right_arrow:WAVAX, xDAI:left_right_arrow:WXDAI, POL/MATIC:left_right_arrow:WMATIC/WPOL) is provided by the implementation. ALWAYS use the configured mapping for the connected chainId.
+NEVER invent or infer a wrapped address.
+Sell native rule (wrap requirement)
+If the sell token is native, the protocol does not support direct sells. ALWAYS inform the user and use the wrapped token for the order; include a wrap step if needed.
+If the buy token is native, CoW supports it directly (no unwrap step required), unless the flow is an explicit unwrap (see equality note below).
+Bridged/aliased variants
+ALWAYS treat bridged variants as distinct (e.g., USDC vs USDC.e, WETH vs WETH.e).
+NEVER collapse bridged tokens into a single address.
```

Comment:
Where did this come from? Why are we adding it here? Have we ever had any issues with "bridged assets"?